cheri/cheri_types.h

```diff
@@ -97,7 +98,8 @@ struct cheri_reg_t {
 
   uint64_t offset() const { return _cursor - _base; }
   cheri_length_t length() const { return _top - _base; }
-  bool sealed() const { return otype != OTYPE_UNSEALED; }
+  bool sealed() const { return ((otype != OTYPE_UNSEALED) && (otype != OTYPE_SENTRY)); }
```

Comment:
This does not seem like a good idea to me. You'll get lots of accidental sentry modifications if you do this.
For almost all instructions sentries should be treated the same way as sealed with type.

Reply:
I agree. It's just that we don't exercise sentries or sealed caps in Spike atm; we just throw a trap in various places/instructions if the capability is `sealed()`. This was the most straightforward/easy change to get recently built binaries working again with `csealentry.` In the future, when we handle sealed caps and sentries, we definitely want to refactor the code.